### DISCUSSION TITLE
Feedback on portal UI deployment

### DISCUSSION THREAD

**Discussion by danycontre** [Azure/avdaccelerator#60] [General]
Creating this issue to consolidate feedback on the custom portal UI:

https://portal.azure.com/#blade/Microsoft_Azure_CreateUIDef/CustomDeploymentBlade/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Favdaccelerator%2Fem-customui%2F01-avd-framework.json/uiFormDefinitionUri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Favdaccelerator%2Fem-customui%2FPortalDefinitions%2F01-landingzone.json

**Comment by danycontre** [comment]
Some initial comments:
- New scaling tab: There should be an scaling tab to let you you choose if using start vm on connect or scaling plan (once it is GA).
- Components tab: depending on the automation option selected the custom role should be created if it doesn't exist and the RBAC permissions assignment created, maybe we could show a message saying this will be done, but don't see a reason for the option since deploying the role is mandatory for the feature to work. (this could be moved to the scaling tap if we create it).
- Components tab: we should have an option to deploy an empty application group for remote apps.
- Host pool settings tab, we could add and advanced option with a text box field to enter specific RDP properties
- Idenity: don't see an option for authentication (ADDS or AADDS), for the MVP we could just show a message about it on the network section. on the network section stating that the subnet hosting the session hosts needs to have line of sight to the identity service (ADDS or AADDS).
- Network: when selecting existing vNet, we should show a message stating that the subnet hosting the session hosts needs to have line of sight to the identity service (ADDS or AADDS).
- Network: when selecting to create a new vNet we should ask for vNet where ADDS or AADDS exist to create peering (we could remove the new vNet option on the MVP).

**Reply by edm-ms** [reply]
- New scaling tab: 

We should put this in the backlog for now

- Components tab custom role: 

We do all this if "create custom role" is selected including scoping the role to the subscription where we deploy. We give the option of creating it or not in the event someone wants to run this accelerator again, or if they already have created the role somewhere else or had someone create it for them.

- Components tab remote app: 

I think we should leave this to another deployment vs the initial landing zone. The reason here is users will need to install apps in an image to publish unless they were only wanting to publish default apps, and we want to reduce the amount of options they need to select to deploy. 

- Host pool settings tab RDP properties: 

This is already there.

- Identity: 

Agreed. I need to build in some logic for AD DS or AADDS. We can prompt to create AADDS if it doesn't exist.

- Network Requirements: 

Agreed and I added a note box explaining this.

- Network Peering: 

The hard part is RBAC. A user doing this deployment may not have write access to the transit VNet since it is likely to live in another subscription. I think we should just make note of the connectivity required, and since we do not deploy a session host as part of the base landing zone deployment nothing would fail (AIB uses an ephemeral VNet to do the initial build).

**Reply by danycontre** [reply]
> Thanks Erick, agree with your comments on getting some things in the backlog and maybe deliver them in the future as add-on modules.
>
> * For the host pool RDP settings wasn't able to find an option in the portal UI to enter custom properties in a text box (i could be looking in the wrong place).
> * Agree network peering will require RBAC setup that will be tricky to manage, so for the MVP we could just tell the user that it will be required prior deploying session hosts.

**Reply by edm-ms** [reply]
Third tab: 

![image](https://user-images.githubusercontent.com/36306721/151191535-4125a067-ee15-4ed3-83f1-e8f752c64a42.png)

The 3 radio buttons at the top.

![image](https://user-images.githubusercontent.com/36306721/151191707-41df4d98-e230-4cdc-a556-7629a72fa867.png)

**Reply by danycontre** [reply]
Hey Erick, when i said custom RDP properties text box i was referring to something similar to what we can see in the host pool properties:

<img width="768" alt="image" src="https://user-images.githubusercontent.com/78437433/151238806-d39b5eec-b71c-48b1-b5d5-afa39f30d676.png">

Not something we really need to have specially for the MVP, but know of multiple customers that already have the string for things they want to set as RDP properties.

**Reply by edm-ms** [reply]
Those are the same properties. What you choose in those boxes will create what you see here. If we need to add another box to copy/paste properties we could expose it though.

**Reply by danycontre** [reply]
It would be nice to have the option, though not really a critical thing, better to work on the top ones and get this one on the backlog for future updates
